lib/pure/sugar.nim

```diff
@@ -238,3 +238,17 @@ macro distinctBase*(T: typedesc): untyped =
   while typeSym.typeKind == ntyDistinct:
     typeSym = getTypeImpl(typeSym)[0]
   typeSym.freshIdentNodes
+
+when defined(nimHasAlias):
+  proc aliasImpl[T1, T2](name: T1, expr: T2) {.magic: "Alias".}
```

Comment:
How is the type checker supposed to resolved T1 here? Isn't T1 just an identifier? I think the reason it does not cause a compilation error is because of of _magic_. But I think ``untyped`` is much proe suitable here instead of ``T1``. ``T2`` is ok though.

Reply:
done. and yes, it worked because it's a `magic`; but I've changed this to use `untyped`